demo/src/main/java/App.java

```diff
@@ -32,6 +32,11 @@ public static void main(String[] args) {
         Pyroscope.setStaticLabels(mapOf("region", "us-east-1"));
 
         appLogic();
+
+        Thread.sleep(2 * 60 * 1000);
+
+        // This is a naive implementation for the stop method
+        PyroscopeAgent.stop();
```

Comment:
To test this properly I suggest, you Start it, then in a separate thread: stop it after a 1 second and then start it again a second later. This will make sure it works as expected.

Reply:
I have added the test. Please re-review. Let me know if I am making any blunders as well